packages/rulesets/src/spectral/functions/put-response-codes.ts

```diff
@@ -0,0 +1,48 @@
+// Synchronous and LR PUT must have 200 & 201 return codes.
+// RPC Code: RPC-Async-V1-01
+
+const LR_AND_SYNC_RESPONSES = ["200", "201", "default"]
+const LR_AND_SYNC_ERROR =
+  "Synchronous and Long-running PUT operations must have responses with 200, 201 and default return codes. They also must not have other response codes."
+const EmptyResponse_ERROR =
+  "PUT operation response codes must be non-empty. It must have response codes 200, 201 and default for both synchronous and long running."
+
+export const PutResponseCodes = (putOp: any, _opts: any, ctx: any) => {
+  if (putOp === null || typeof putOp !== "object") {
+    return []
+  }
+  const path = ctx.path
+  const errors = []
+
+  const responses = Object.keys(putOp?.responses ?? {})
+
+  if (responses.length == 0) {
+    errors.push({
+      message: EmptyResponse_ERROR,
+      path: path,
+    })
+    return errors
+  }
+
+  const isAsyncOperation = (putOp["x-ms-long-running-operation"] && putOp["x-ms-long-running-operation"] === true) || !! putOp["x-ms-long-running-operation-options"]
```

Comment:
>!! [](http://example.com/codeflow?start=118&length=2)

?

Reply:
Ah, I forgot to remove these
I added them so that isAsyncOperation would return a bool and not an object. However it works either way.

Ill get rid of it.